src/otaclient/ota_core.py

```diff
@@ -566,14 +581,30 @@ def _execute_update(self):
         )
 
         try:
-            delta_calculator = DeltaGenerator(
-                ota_metadata=self._ota_metadata,
-                delta_src=Path(cfg.ACTIVE_SLOT_MNT),
-                copy_dst=self._resource_dir_on_standby,
-                status_report_queue=self._status_report_queue,
-                session_id=self.session_id,
-            )
-            delta_calculator.calculate_delta()
+            if check_base_filetable(_base_ft_db):
+                logger.info(
+                    f"file_table for active_slot({_base_ft_db}) found and valid, use file_table to assist delta calculation!"
+                )
+                delta_calculator = DeltaGenWithFileTable(
+                    ota_metadata=self._ota_metadata,
+                    delta_src=Path(cfg.ACTIVE_SLOT_MNT),
+                    copy_dst=self._resource_dir_on_standby,
+                    status_report_queue=self._status_report_queue,
+                    session_id=self.session_id,
+                )
+                delta_calculator.calculate_delta(base_file_table=_base_ft_db)
+            else:
+                logger.info(
+                    f"file_table for active_slot({_base_ft_db}) not found/invalid, use full disk scan for delta calculation!"
+                )
+                delta_calculator = DeltaGenFullDiskScan(
+                    ota_metadata=self._ota_metadata,
+                    delta_src=Path(cfg.ACTIVE_SLOT_MNT),
+                    copy_dst=self._resource_dir_on_standby,
+                    status_report_queue=self._status_report_queue,
+                    session_id=self.session_id,
+                )
+                delta_calculator.calculate_delta()
```

Comment:
Explicitly choose which DeltaGenerator to use on whether the file_table database is available or not.